app/src/main/java/com/example/indoor_volleyball/Fragments/ProfileFragment.java

```diff
@@ -174,12 +177,13 @@ public void displayUserInfo() {
         binding.tvUsername.setText(user.getUsername());
         ParseFile profilePhoto = user.getParseFile("profilePhoto");
         if (profilePhoto != null) {
-            Glide.with(requireContext()).load(user.getParseFile("profilePhoto").getUrl()).circleCrop().into(binding.ivProfilePhoto);
+            Glide.with(requireContext()).load(Objects.requireNonNull(user.getParseFile("profilePhoto")).getUrl()).circleCrop().into(binding.ivProfilePhoto);
         } else {
             Toast.makeText(getContext(), "this Profile photo does not exist for user " + user.getUsername(), Toast.LENGTH_SHORT).show();
         }
     }
 
+    //I could not find a non depreciated version.
```

Comment:
This is what you want to use one of those activity result launchers for.  THat's the new way of doing this.  There is probably a take picture launcher already built for you.  

EDIT:  Yup!  there is.  https://www.daniweb.com/programming/mobile-development/tutorials/536924/android-native-take-a-picture-with-activityresult-api